### PR TITLE
Add flash-varlen, CLD, TIDE

### PR DESCRIPTION
**Pull request by trahxam**
closes https://github.com/samvanstroud/hepattn/issues/13

### CODE REVIEW

File: src/hepattn/utils/cli.py [hunk 1]
```diff
@@ -7,10 +7,17 @@
 from jsonargparse.typing import register_type
 from lightning.pytorch.cli import LightningCLI
 
+
+# For info on why this is needed, see here
+# https://docs.pytorch.org/docs/stable/multiprocessing.html#file-system-file-system
+torch.multiprocessing.set_sharing_strategy("file_system")
```
Comment (samvanstroud) [inline src/hepattn/utils/cli.py:13]: I had a quick look at the link but not sure exactly why this is needed?

File: src/hepattn/models/loss.py [hunk 2]
```diff
@@ -34,12 +34,22 @@ def mask_dice_loss(pred_logits, true, mask=None, weight=None):
 
 def mask_dice_costs(pred_logits, true):
     pred = pred_logits.sigmoid()
-    num = 2 * torch.einsum("bnc,bmc->bnm", pred, true)
-    den = pred.sum(-1).unsqueeze(2) + true.sum(-1).unsqueeze(1)
-    losses = 1 - (num + 1) / (den + 1)
+    intersection = 2 * torch.einsum("bnc,bmc->bnm", pred, true)
+    num_pred = pred.sum(-1).unsqueeze(2)
+    num_true = true.sum(-1).unsqueeze(1)
+    losses = 1 - (2 * intersection + eps) / (num_pred + num_true + eps)
```
[the remark this answers is not in the document]
Comment (trahxam) [reply src/hepattn/models/loss.py:40]: Was just cleaning up names while trying to get dice loss to work. I also tried IOU hence the names.

[the remark this answers is not in the document]
Comment (trahxam) [reply src/hepattn/models/loss.py:40]: Ah yeah my bad, the extra two is a typo.
